### DISCUSSION TITLE
$nextTick for Livewire/AlpineJS integration

### DISCUSSION THREAD

**Discussion by ju5t** [livewire/livewire#2011] [1. Help]
We have the following code in a Livewire component:

```
@if ($selected)
<span>
    {{ $selected['name'] }}
    <div>
        <svg x-on:click="$wire.resetSelectedId()" class="cursor-pointer h-5 w-5 text-red-500" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
            <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z" />
        </svg>
    </div>
</span>
@else
<x-input.text x-on:focus="showResults = true" x-on:click.away="showResults = false" x-ref="query" wire:model="query" class="form-select" />
@endif
```

`$selected` is a model, a customer for example. The if-statement checks if it's not `null`.

Assuming the current state of `$selected` is a model, a customer. The first if-statement equals true and a static field renders with the `name` of the `customer`. This all makes sense. If a user clicks the `x` (that's the svg) we reset `selected` and the input fields is shown. This is as expected.

But, if the user clicks the `x`, I'd like the input field to gain focus. This is what `$nextTick` would do, but not if coupled with Livewire.

What would be a good solution for this that isn't `setTimeout()`? We can't use x-init to always set focus, it only has to happen when someone clicks the `x`, as we would have multiple input fields like these.

**Comment by joshhanley** [comment]
I would probably control the visibility using alpine and then show and focus once the `x` button has been selected rather than putting it in a blade conditional.

Hope this helps!
